File: backend/internal/models/recipe.go
package models

import "time"

type Recipe struct {
	ID          int64     `db:"id"          json:"id"`
	Title       string    `db:"title"        json:"title"`
	Description string    `db:"description"  json:"description"`
	CuisineID   int       `db:"cuisine_id"   json:"cuisine_id"`
	CuisineName string    `db:"cuisine_name" json:"cuisine_name"`
	CookTime    int       `db:"cook_time"    json:"cook_time"`
	Servings    int       `db:"servings"     json:"servings"`
	Difficulty  int       `db:"difficulty"   json:"difficulty"`
	CreatedAt   time.Time `db:"created_at"   json:"created_at"`
}

type CreateRecipeRequest struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	CuisineID   int    `json:"cuisine_id"`
	CookTime    int    `json:"cook_time"`
	Servings    int    `json:"servings"`
	Difficulty  int    `json:"difficulty"`
}

type UpdateRecipeRequest struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	CookTime    int    `json:"cook_time"`
	Servings    int    `json:"servings"`
	Difficulty  int    `json:"difficulty"`
}

type ListResponse struct {
	Data       []Recipe `json:"data"`
	Total      int64    `json:"total"`
	NextCursor int64    `json:"next_cursor"`
}

type Ingredient struct {
	Name   string  `db:"name"   json:"name"`
	Amount float64 `db:"amount" json:"amount"`
	Unit   string  `db:"unit"   json:"unit"`
}

File: ios/RecipeApp/RecipeApp/Models/Recipe.swift

import Foundation

struct Recipe: Identifiable, Codable {
    let id: Int
    let title: String
    let description: String
    let cuisineId: Int
    let cuisineName: String
    let cookTime: Int
    let servings: Int
    let difficulty: Int
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case description
        case cuisineId    = "cuisine_id"
        case cuisineName  = "cuisine_name"
        case cookTime     = "cook_time"
        case servings
        case difficulty
        case createdAt    = "created_at"
    }
}

struct RecipeListResponse: Codable {
    let data: [Recipe]
    let nextCursor: Int

    enum CodingKeys: String, CodingKey {
        case data
        case nextCursor = "next_cursor"
    }
}

struct CreateRecipeRequest: Codable {
    let title: String
    let description: String
    let cuisineId: Int
    let cookTime: Int
    let servings: Int
    let difficulty: Int

    enum CodingKeys: String, CodingKey {
        case title
        case description
        case cuisineId  = "cuisine_id"
        case cookTime   = "cook_time"
        case servings
        case difficulty
    }
}

struct Ingredient: Codable, Identifiable {
    var id: String { name }
    let name: String
    let amount: Double
    let unit: String
}

struct IngredientsResponse: Codable {
    let data: [Ingredient]
}